R/coerce_to_messydate.R

```diff
@@ -3,6 +3,15 @@
 #' These functions coerce different data classes into `messydt` class
 #' @param x A scalar or vector of a class that can be coerced into a Date,
 #' such as `Date`, `POSIXct`, `POSIXlt`, or character.
+#' @param text Do you want dates to be extract dates from text?
+#' By default false.
+#' If TRUE, dates are extracted from text.
+#' If multiple dates are identified in the same row,
+#' only the first date is returned.
+#' @param interactive By default FALSE.
+#' If TRUE, it allows users to choose the correct order
```

Comment:
One thing I thought of is that if it is a column in a dataframe with many dates and only some of them are ambiguous 6-digit dates in the wrong order, is there a way the user can select just one option for all the dates (eg. M-Y-D) and then these dates get automatically sorted? Right now I think it works by either applying the correct order only on the first date in the column or the user has to go through the dates one row at a time (which would be tedious if it's a large dataset). Maybe this is where an independent resequence function might come into play as @henriquesposito proposed?

Reply:
Thank you for the comment Jael. In theory, the selected format order is passed through all ambiguous 6 digit dates in a vector and not anymore one by one. For example you run:

```
as_messydate(c("10-11-12", "10-11-12", "2012-11-10"), interactive = TRUE)
# and select DMY as the answer, the order of the first two dates will be changed to DMY.
``` 

However, I understand where your question is coming from and I just wrangle the question to make it more clear to users. Please let me know if the new wording makes sense.  

I also like your idea of repurposing resequence here, maybe instead of interactive we can call the argument resequence and expand on what options users can choose?

Reply:
I'd like to see what this would look like. Why don't you (perhaps together) work on a proposal?

Reply:
We could change the argument `interactive` to `resequence` to make it more intuitive for users that the function helps to reorder date components in ambiguous 6 digit dates. We could also extend `resequence` to complete the 6 digit dates into the standard YYYY-MM-DD format after reordering the date components according to users' input, instead of leaving it as a 6 digit date.

As an example,
`as_messydate(c("09-11-12", "08-11-12", "2012-11-10"), resequence = TRUE)`
`# if option 2: DMY (Day-Month-Year) is chosen, output would be something like this:`
`       dates`
`1 2012-11-09`
`2 2012-11-08`
`3 2012-11-10`

@jhollway please let us know what you think, thank you!

Reply:
Thank you @jaeltan ! I agree that changing the argument name to resenquence, instead of interactive, makes it more intuitive for users. I also like the idea of giving user's the choice to complete, if they want to,  and how six digit dates. I will work on something and we can see if it works well, or not. Thank you!

Reply:
Tentatively, I have changed the argument name from interactive to "resequence" and also added a new section about completing 6 digit dates, if users want to. This is a first draft and, if we choose to go this route I can make it more flexible so that users have more options or can complete with the value they want. We could also make resequence, as it stands, as a separate function. Again, this is tentatively and we can go back to what we had before or take it in other directions. @jhollway and @jaeltan please let me know what you think. Thank you.

Reply:
So how does it work @henriquesposito ? Should the function report where the ambiguity is (or at least the first instance), what compatible dates/options there are, and then warn the user that this will be used for all other such ambiguous dates?

Reply:
As it stands, If resequence is set to TRUE, the function works by asking users about the order of ambiguous 6 digits dates as well as if all 6 digit dates for which the year is between 0 and 22 are in the 21st century and if all 6 digit dates for which the year is bigger than 22 in the 20th century. If users respond yes, the function completes these dates, otherwise these stay the same.

Once again, this is just tentatively and if we choose to proceed in this direction we can offer users more flexibility to complete these dates or, even, offer this version of resequence as a separate function.